lib/service/invoices.go

```diff
@@ -223,11 +224,33 @@ func (svc *LndhubService) HandleFailedPayment(ctx context.Context, invoice *mode
 	return err
 }
 
-func (svc *LndhubService) HandleSuccessfulPayment(ctx context.Context, invoice *models.Invoice) error {
+func (svc *LndhubService) HandleSuccessfulPayment(ctx context.Context, invoice *models.Invoice, parentEntry models.TransactionEntry) error {
+	// Get the user's fee account for the transaction entry, current account is already there in parent entry
+	feeAccount, err := svc.AccountFor(ctx, common.AccountTypeFees, invoice.UserID)
+	if err != nil {
+		svc.Logger.Errorf("Could not find fees account user_id:%v", invoice.UserID)
+		return err
+	}
+	// add transaction entry for fee
+	entry := models.TransactionEntry{
+		UserID:          invoice.UserID,
+		InvoiceID:       invoice.ID,
+		CreditAccountID: feeAccount.ID,
+		DebitAccountID:  parentEntry.DebitAccountID,
+		Amount:          int64(invoice.Fee),
+		ParentID:        parentEntry.ID,
+	}
+	_, err = svc.DB.NewInsert().Model(&entry).Exec(ctx)
+	if err != nil {
+		sentry.CaptureException(err)
+		svc.Logger.Errorf("Could not insert fee transaction entry user_id:%v invoice_id:%v", invoice.UserID, invoice.ID)
+		return err
+	}
+
 	invoice.State = common.InvoiceStateSettled
 	invoice.SettledAt = schema.NullTime{Time: time.Now()}
 
-	_, err := svc.DB.NewUpdate().Model(invoice).WherePK().Exec(ctx)
+	_, err = svc.DB.NewUpdate().Model(invoice).WherePK().Exec(ctx)
```

Comment:
if we fail to save the transaction entry the invoice will not be update. 
I think it makes sense to first save the invoice and then save the fee transaction entry. 
This way we do not loose data (preimage and that the invoice was actually paid) on the invoice if we somehow fail to save the transaction entry (unlikely though)